internal/cluster/distribution/eks/eksworkflow/activity_update_addon.go

```diff
@@ -0,0 +1,187 @@
+		}
+		return nil, errors.WrapIfWithDetails(err, "failed to retrieve addon", "cluster", input.ClusterName, "addon", input.AddonName)
+	}
+
+	currentVersion := *addonOutput.Addon.AddonVersion
+	logger.Infof("addon current version: %v", currentVersion)
+
+	describeAddonVersionInput := &eks.DescribeAddonVersionsInput{
+		AddonName:         aws.String(input.AddonName),
+		KubernetesVersion: aws.String(input.KubernetesVersion),
+	}
+	addonVersionsOutput, err := eksSvc.DescribeAddonVersions(describeAddonVersionInput)
+	if err != nil {
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) {
+			err = errors.New(awsErr.Message())
+		}
+		return nil, errors.WrapIfWithDetails(err, "failed to retrieve addon versions", "cluster", input.ClusterName, "addon", input.AddonName)
+	}
+
+	selectedVersion, err := selectLatestVersion(addonVersionsOutput, currentVersion, input.KubernetesVersion)
+	if err != nil {
+		return nil, errors.WrapIfWithDetails(err, "error selecting new version", "cluster", input.ClusterName, "addon", input.AddonName)
+	}
+	if selectedVersion == currentVersion {
+		logger.Infof("no newer version available then current version: %s", currentVersion)
+		return &UpdateAddonActivityOutput{UpdateID: ""}, nil
+	}
+
+	logger.Infof("update addon to selected version: %v", selectedVersion)
+	updateAddonInput := &eks.UpdateAddonInput{
+		AddonName:        aws.String(input.AddonName),
+		ClusterName:      aws.String(input.ClusterName),
+		ResolveConflicts: aws.String(eks.ResolveConflictsOverwrite),
+		AddonVersion:     aws.String(selectedVersion),
+	}
+	updateAddonOutput, err := eksSvc.UpdateAddon(updateAddonInput)
+	if err != nil {
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) {
+			err = errors.New(awsErr.Message())
+		}
+		return nil, errors.WrapIfWithDetails(err, "failed to update addon", "cluster", input.ClusterName, "addon", input.AddonName)
+	}
+	output := UpdateAddonActivityOutput{UpdateID: aws.StringValue(updateAddonOutput.Update.Id)}
+
+	return &output, nil
+}
+
+func selectLatestVersion(addonVersions *eks.DescribeAddonVersionsOutput, currentVersion string, kubernetesVersion string) (string, error) {
+	currentVersionSemver, err := semver.NewVersion(currentVersion)
+	if err != nil {
+		return "", err
+	}
+	latestVersion := currentVersionSemver
+
+	for _, addon := range addonVersions.Addons {
+		for _, version := range addon.AddonVersions {
+			if !versionIsCompatible(version.Compatibilities, kubernetesVersion) {
+				continue
+			}
+			newVersion, err := semver.NewVersion(*version.AddonVersion)
+			if err != nil {
+				return "", err
+			}
+			if newVersion.GreaterThan(latestVersion) {
+				latestVersion = newVersion
+			}
+		}
+	}
+	return latestVersion.Original(), nil
+}
+
+func versionIsCompatible(compatibilities []*eks.Compatibility, kubernetesVersion string) bool {
+	for _, c := range compatibilities {
+		if c.ClusterVersion == nil {
+			continue
+		}
+		if kubernetesVersion == *c.ClusterVersion {
```

Comment:
**Recommended**: IMO the [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions) **SHOULD** be replaced to reduce [cognitive load](https://en.wikipedia.org/wiki/Cognitive_load), because `kubernetesVersion` is a dynamically initialized constant throughout the function and `c` is the iteration variable.

```suggestion
		if *c.ClusterVersion == kubernetesVersion {
```